### PR TITLE
Add Compatibility Notes to RELEASES.md for 1.38.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/rust/blob/relnotes/RELEASES.md)

r? @Mark-Simulacrum

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -68,6 +68,13 @@ Misc
 - [`rustc` will now warn about some incorrect uses of
   `mem::{uninitialized, zeroed}` that are known to cause undefined behaviour.][63346]
 
+Compatibility Notes
+-------------------
+- Unfortunately the [`x86_64-unknown-uefi` platform can not be built][62785]
+  with rustc 1.39.0.
+- The [`armv7-unknown-linux-gnueabihf` platform is also known to have
+  issues][62896] for certain crates such as libc.
```
Comment (jonas-schievink) [inline RELEASES.md]: Missing issue/PR references

**Comment by jonas-schievink** [comment]
The rendered link points to the wrong branch

**Comment by XAMPPRocky** [comment]
It also wasn't for 1.39.0. It's late.

**Comment by Mark-Simulacrum** [comment]
@bors r+ p=1

(It's mostly important that this gets into beta, though, so this landing in master is not super high priority. Can be rolled up).

**Pull request merged**